prestodb/transaction.py

```diff
@@ -0,0 +1,74 @@
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+import logging
+
+from prestodb import constants
+import prestodb.exceptions
+
+
+logger = logging.getLogger(__name__)
+
+
+class Transaction(object):
+    def __init__(self, request):
+        self._request = request
+        self._id = 'NONE'
+
+    @property
+    def id(self):
+        return self._id
+
+    def begin(self):
+        response = self._request.post('START TRANSACTION')
+        if response.status_code != self._request.http.codes.ok:
+            raise prestodb.exceptions.DatabaseError(
+                'failed to start transaction: {}'.format(response.status_code))
+        status = self._request.process(response)
+        if not status.next_uri:
+            self._id = (
+                response.headers[constants.HEADER_STARTED_TRANSACTION]
+            )
+            return
```

Comment:
Parentheses are only here to wrap the line. There is no comma before the closing parenthesis, hence it is a scalar value, not a *tuple*. I note that it can create confusion though.